src/main/scala/chisel3/package.scala

```diff
@@ -245,6 +245,8 @@ package object chisel3 {    // scalastyle:ignore package.object.name
 
   type BlackBox = chisel3.core.BlackBox
 
+  type InstanceId = chisel3.internal.InstanceId
```

Comment:
This is a question for @jackkoenig if this should be exposed outside of internal. I've had to use this for external libraries, so it likely makes sense.

Reply:
I think InstanceId is the wrong API and I wish it weren't necessary to expose it but it's obviously the only API so what can we do.

That being said, I think this PR should not be using InstanceId, see my comment below on the ChiselAnnotation

Reply:
Got it. Thanks for figuring out a way around this.

Reply:
Since the PR no longer needs InstanceId, should we still be doing this or have a separate PR more focused on that question?